### DISCUSSION TITLE
back-up save games: cross platform?

### DISCUSSION THREAD

**Discussion by ghost** [SuperTux/supertux#2874] [Q&A]
As asked here https://github.com/SuperTux/supertux/issues/2871

If I make a back-up of a profile folder, is it cross-platform?
I use supertux on windows and android and would like to copy progress between those versions (because I play then on windows and then on android).
Is that possible or not recommended?

**Comment by provigz** [comment]
> If I make a back-up of a profile folder, is it cross-platform?

Yes, it will work anywhere, as long as the SuperTux version is the same.

> I use supertux on windows and android and would like to copy progress between those versions

The latest official v0.6.3 Android build, I believe, requires rooting your Android device to access/modify user data, in which the profile is stored, so I'd say it's not recommended.

However, the WIP revamp of Android builds, planned to be ready for the next release, should allow easier access to this user data folder, hopefully making this easier in the future.

**Comment by ghost** [comment]
> However, the WIP revamp of Android builds, planned to be ready for the next release, should allow easier access to this user data folder, hopefully making this easier in the future.

I hope it's ready soon! Please allow us to access the folder without rooting! 
Is there already plans for the new release? It has been since 2021 now.

**Reply by provigz** [reply]
Planned to release around August, 2024.

**Comment by ghost** [comment]
2 more questions:

- Is it correct to backup all game progress, I just need to copy the whole folder "profile1" (or any other profile I use) or are additional folders also needed?
- Is it correct each .stsg file is a save game file?

**Reply by provigz** [reply]
> Is it correct to backup all game progress, I just need to copy the whole folder "profile1" (or any other profile I use) or are additional folders also needed?

It depends on whether you want to backup only one or a few profiles, or all of them. You can see which profile you're on in "Options" -> "Select Profile".

> Is it correct each .stsg file is a save game file?

Yes, each is specific to a levelset.

**Comment by ghost** [comment]
> It depends on whether you want to backup only one or a few profiles, or all of them. You can see which profile you're on in "Options" -> "Select Profile".

OK. A profile is nothing more then another way to save progress of another user for example? So the progress is separated by player?

**Reply by provigz** [reply]
> OK. A profile is nothing more then another way to save progress of another user for example? So the progress is separated by player?

Yes. You can use it for separating progress for multiple players.

---

Also, when using GitHub Discussions, to reply to a thread, instead of creating a new one, press the little "Write a reply" textbox under it instead.

**Reply by ghost** [reply]
Oops! Yes, sorry, I'm new everywhere, also on github :) Everything is clear now, I close it :)
